core/trino-main/src/main/java/io/trino/execution/scheduler/policy/SequentialExecutionSchedule.java

```diff
@@ -0,0 +1,606 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution.scheduler.policy;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.trino.execution.StateMachine.StateChangeListener;
+import io.trino.execution.scheduler.ExecutionSchedule;
+import io.trino.execution.scheduler.PipelinedStageExecution;
+import io.trino.execution.scheduler.PipelinedStageExecution.State;
+import io.trino.sql.planner.PlanFragment;
+import io.trino.sql.planner.plan.AggregationNode;
+import io.trino.sql.planner.plan.ExchangeNode;
+import io.trino.sql.planner.plan.IndexJoinNode;
+import io.trino.sql.planner.plan.JoinNode;
+import io.trino.sql.planner.plan.PlanFragmentId;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.PlanVisitor;
+import io.trino.sql.planner.plan.RemoteSourceNode;
+import io.trino.sql.planner.plan.SemiJoinNode;
+import io.trino.sql.planner.plan.SpatialJoinNode;
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.EdgeFactory;
+import org.jgrapht.alg.StrongConnectivityInspector;
+import org.jgrapht.graph.DefaultDirectedGraph;
+import oshi.annotation.concurrent.GuardedBy;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.execution.scheduler.PipelinedStageExecution.State.FLUSHING;
+import static io.trino.execution.scheduler.PipelinedStageExecution.State.RUNNING;
+import static io.trino.execution.scheduler.PipelinedStageExecution.State.SCHEDULED;
+import static io.trino.sql.planner.plan.AggregationNode.Step.FINAL;
+import static io.trino.sql.planner.plan.AggregationNode.Step.SINGLE;
+import static io.trino.sql.planner.plan.ExchangeNode.Scope.LOCAL;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+
+/**
+ * Schedules stages in sequence in order to provide the best resource utilization.
```

Comment:
"Schedules stages in sequence" is not meaningful. Every execution policy selects some "sequence".

Phrase it just "Schedules stages choosing to order to provide best resource utilization"

Maybe we should also change the execution policy name from "sequential" to sth more descriptive:
 `prioritize-utilization` maybe?

Reply:
> Maybe we should also change the execution policy name from "sequential" to sth more descriptive:
prioritize-utilization maybe?

Thanks. I was looking for better name